R/plotBoxplot.R

```diff
@@ -442,6 +483,39 @@ setMethod("plotBoxplot", signature = c(object = "SummarizedExperiment"),
     df <- df |>
         as.data.frame()
 
+    # Add p-values placement
+    if( !is.null(p.value) ){
+        p.value <- .add_p_value_position(p.value, x, c(assay.type, col.var, row.var), group.by, fill.by, facet.by, df, ...)
+    }
+
+    # Check that p.value data.frame has the correct grouping variables. If
+    # group.by or fill.by were specified the comparisons were made based on
+    # them. If they are not specified, the comparison were made based on x axis
+    # variable. Check that this is correct.
+    if( !is.null(p.value) && length(c(group.by, fill.by)) > 0L &&
+            !all(c(p.value[["group1"]], p.value[["group2"]]) %in%
+                df[[unique(fill.by, group.by)]]) ){
+        stop("Groups in p.value[['group1']] and p.value[['group2']] ",
+            "must match with 'fill.by'/'group.by'.", call. = FALSE)
+    } else if( !is.null(p.value) && length(c(group.by, fill.by)) == 0L &&
+            !all(c(p.value[["group1"]], p.value[["group2"]]) %in% df[[x]]) ){
+        stop("Groups in p.value[['group1']] and p.value[['group2']] ",
+            "must match with 'x'.", call. = FALSE)
+    }
+    # If grouping variables were specified, x was used as group_by variable and
+    # not as comparison variable.
+    if( !is.null(p.value) && !is.null(x) && !all(p.value[[x]] %in% df[[x]])){
+        stop("Groups in p.value[['", x, "']] ",
+            "must match with 'x'.", call. = FALSE)
+    }
+    # If facetting is specified in the plot, the same facet variable should be
+    # found from p values.
+    if( !is.null(facet.by) && !is.null(p.value) &&
+            !all(p.value[[facet.by]] %in% df[[facet.by]]) ){
+        stop("Groups in p.value[['", facet.by, "']] ",
+            "must match with 'facet.by'.", call. = FALSE)
+    }
```

Comment:
These could be moved to input check. The p value table calculated by the method should be ok.